src/vivarium_gates_mncnh/components/mortality.py

```diff
@@ -163,3 +166,120 @@ def get_proportional_case_fatality_rates(self, simulants: pd.DataFrame) -> pd.Da
+    #####################
+
+    def setup(self, builder: Builder) -> None:
+        self._sim_step_name = builder.time.simulation_event_name()
+        self.randomness = builder.randomness.get_stream(self.name)
+        self.location = get_location(builder)
+
+        self.probability_death_by_age_group = builder.value.register_value_producer(
+            "probability_death_by_age_group",
+            source=self.lookup_tables["probability_death_by_age_group"],
+            component=self,
+            requires_columns=get_lookup_columns(
+                [self.lookup_tables["probability_death_by_age_group"]]
+            ),
+        )
+
+    def on_initialize_simulants(self, pop_data: SimulantData) -> None:
+        pop_update = pd.DataFrame(
+            {
+                COLUMNS.CHILD_CAUSE_OF_DEATH: "not_dead",
+                COLUMNS.CHILD_YEARS_OF_LIFE_LOST: 0.0,
+            },
+            index=pop_data.index,
+        )
+        self.population_view.update(pop_update)
+
+    def on_time_step(self, event: Event) -> None:
+        if self._sim_step_name() not in [
+            SIMULATION_EVENT_NAMES.EARLY_NEONATAL_MORTALITY,
+            SIMULATION_EVENT_NAMES.LATE_NEONATAL_MORTALITY,
+        ]:
+            return
+
+        pop = self.population_view.get(event.index)
+        alive_children = pop.loc[pop[COLUMNS.CHILD_ALIVE] == "alive"]
+
+        # Update age of children to get correctlookup values
+        if self._sim_step_name() == SIMULATION_EVENT_NAMES.EARLY_NEONATAL_MORTALITY:
+            pop.loc[alive_children.index, COLUMNS.CHILD_AGE] = 7 / 365.0
+        else:
+            pop.loc[alive_children.index, COLUMNS.CHILD_AGE] = 28 / 365.0
+        self.population_view.update(pop)
+
+        # Determine which neonates die and update metadata
+        dead_idx = self.randomness.filter_for_rate(
+            alive_children.index,
+            self.probability_death_by_age_group(alive_children.index),
+            f"{self._sim_step_name}_neonatal_mortality_choice",
+        )
+        if not dead_idx.empty:
+            pop.loc[dead_idx, COLUMNS.CHILD_ALIVE] = "dead"
+            pop.loc[dead_idx, COLUMNS.CHILD_CAUSE_OF_DEATH] = "other_causes"
+            pop.loc[dead_idx, COLUMNS.CHILD_YEARS_OF_LIFE_LOST] = self.lookup_tables[
+                "life_expectancy"
+            ](dead_idx)
+
+        self.population_view.update(pop)
+
+    ##################
+    # Helper methods #
+    ##################
+
+    def load_probability_death_by_age_group(self, builder: Builder) -> pd.DataFrame:
+        """Load all-cause mortality rate data."""
+        acmr = builder.data.load("cause.all_causes.cause_specific_mortality_rate")
+        child_acmr = acmr.rename(columns=CHILD_LOOKUP_COLUMN_MAPPER)
+        # Scale by duration of age group
+        child_acmr["value"] = child_acmr["value"] * (
+            child_acmr["child_age_end"] - child_acmr["child_age_start"]
+        )
+        # Convert rate to probability
```

Comment:
Sorry if I caused any confusion here, but I believe `filter_for_rate` already does this part